### PR TITLE
Transit rewrap

### PR DESCRIPTION
**Pull request by yhyakuna**
Adding 2 new guides:

- [Vault Deployment Reference Architecture](https://github.com/hashicorp/vault/blob/1c70a93e80d650c4652d79b74958f56bf9516213/website/source/guides/operations/reference-architecture.html.md) based on [Phil's doc](https://docs.google.com/document/d/1CCbAQ-ZEeSpxQqvo40hM0Y5oe3tBCxM1qYrk7_Wyvww/edit?ts=5a865277#)
- [Transit Secrets Re-wrapping](https://github.com/hashicorp/vault/blob/7bc00dc72f5ba1c1402223b5a92451e821684068/website/source/guides/encryption/transit-rewrap.html.md) based on [Ehron's guide](https://github.com/hashicorp/vault-guides/pull/8) 

Other fixes:
- References to secret **backend** is now secret **engine**
- References to auth **backend** is now auth **method**
- Diagram for AppRole Pull Authentication workflow was updated slightly based on SE's input

### CODE REVIEW

File: website/source/guides/operations/reference-architecture.html.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+engine configuration, and some audit logging configurations) and responses
+consume network bandwidth.
+
+## Other Considerations
+
+[Vault Production Hardening Recommendations](/guides/operations/production.html).
+
+### Network Connectivity Details
+
+![Network Connectivity Details](/assets/images/vault-ref-arch.png)
```
[the remark this answers is not in the document]
Comment (yhyakuna) [reply website/source/guides/operations/reference-architecture.html.md]: updated the diagram

File: website/source/guides/operations/reference-architecture.html.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+Vault disaster recovery replication ensures that a standby Vault cluster is kept
+synchronized with an active Vault cluster.  This mode of replication includes
+data such as ephemeral authentication tokens, time-based token information as
+well as token usage data. This provides for aggressive recovery point objective
+in environments where high availability is of the utmost concern.
+
+#### Diagrams
+
+The following diagrams illustrate some possible replication scenarios.
```
[the remark this answers is not in the document]
Comment (yhyakuna) [reply website/source/guides/operations/reference-architecture.html.md]: diagram has been updated

File: website/source/guides/identity/authentication.html.md [hunk 3]
```diff
@@ -10,27 +10,27 @@ description: |-
...
+method**](/docs/auth/index.html) to acquire a token. This token has policies attached so
 that the behavior of the client can be governed.
 
 Since tokens are the core method for authentication within Vault, there is a
-**token** auth backend (often refer as **_token store_**). This is a special
-auth backend responsible for creating and storing tokens.
+**token** auth method (often refer as **_token store_**). This is a special
+auth method responsible for creating and storing tokens.
 
 ### Auth Backends
```
Comment (topfunky) [inline website/source/guides/identity/authentication.html.md]: Should this be titled "Auth Methods" since the rest of the terminology changed?

Comment (yhyakuna) [reply website/source/guides/identity/authentication.html.md]: I thought I fixed them all...but I must have missed those.  Thanks!

File: website/source/guides/encryption/index.html.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  The transit secrets engine handles cryptographic functions on data in-transit.
+  Vault doesn't store the data sent to the secrets engine. It can also be viewed
+  as "cryptography as a service" or "encryption as a service".
+---
+
+# Encryption as a Service
+
+Vault provides Encryption as a Service (EaaS) to enables security teams to
+fortify data during transit and at rest. So even if an intrusion occurs, your
+data is encrypted with AES 256-bit CBC encryption (TLS in transit). Even if an
```
Comment (vishalnayak) [inline website/source/guides/encryption/index.html.md]: Transit supports multiple cipher types (https://www.vaultproject.io/api/secret/transit/index.html#type). The default is the AES 256-bit GCM 96-bit. But it doesn't support AES in CBC mode.

Comment (yhyakuna) [reply website/source/guides/encryption/index.html.md]: @vishalnayak  Thank you for the correction.  I think I copied that from some old doc.  I updated it to keep the intro simple.

Comment (vishalnayak) [reply website/source/guides/encryption/index.html.md]: No problem!

**Pull request merged**